MathLib/LinAlg/PETSc/PETScLinearSolver.cpp

```diff
@@ -21,29 +21,42 @@
 
 namespace MathLib
 {
-PETScLinearSolver::PETScLinearSolver(const std::string prefix,
+PETScLinearSolver::PETScLinearSolver(const std::string /*prefix*/,
                                      BaseLib::ConfigTree const* const option)
-    : _elapsed_ctime(0.)
 {
-    // Insert options into petsc database if any.
-    if (option)
-    {
+    // Insert options into petsc database. Default options are given in the string below.
+    std::string petsc_options
+            = "-ksp_type cg -pc_type bjacobi -ksp_rtol 1e-16 -ksp_max_it 10000";
```

Comment:
Would like to use bicg as default because `` MatZeroRows() `` is used for applying Dirichlet BC. Only for some simple case, cg can give converged solution. Or we use more complicated cg setting for `` MatZeroRows() ``
``` 
 -ksp_type preonly -pc_type redistribute -redistribute_ksp_type cg -redistribute_pc_type bjacobi -redistribute_sub_pc_type icc 
```
In future, we could consider an option for either using `` MatZeroRows() `` or ``MatZeroRowsColumns()  ``